src/panel/indexStore.spec.ts

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// Exceptions to make mocking easier.
+/* eslint-disable @typescript-eslint/ban-types */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable no-console */
+
+/// <reference path="../panel/global.d.ts" />
+
+import { IndexStore } from './indexStore';
+import { filtersColumn, filtersRow } from '../shared';
+import { Log } from 'sarif';
+import assert from 'assert';
+import { RowItem } from './tableStore';
+
+describe('IndexStore', () => {
+    before('before all', () => {
+        const log = {
+            version: '2.1.0',
+            runs: [{
+                tool: {
+                    driver: { name: 'Driver' }
+                },
+                results: [{
+                    message: {
+                        text: 'Message 1'
+                    },
+                    locations: [{
+                        physicalLocation: {
+                            artifactLocation: {
+                                uri: 'file:///c%3A/Users/muraina/sarif-tutorials/samples/results.sarif',
+                            },
+                            region: {
+                                startLine: 28,
+                            },
+                        }
+                    }]
+                }]
+            }]
+        } as Log;
+        global.fetch = async () => ({ json: async () => log }) as unknown as Promise<Response>;
+        global.vscode = {
+            postMessage: async () => {}
+        };
+    });
+    it('keep logs in sync', async () => {
+        const indexStore = new IndexStore({filtersRow, filtersColumn});
+        // Adds a new log
+        await indexStore.onMessage({ data: {
+            'command': 'spliceLogs',
+            'removed': [],
+            'added': [{
+                'uri':'file:///c%3A/Users/muraina/sarif-tutorials/samples/results.sarif',
+                'webviewUri':'vscode-webview-resource://bd904178-42ae-4a87-aa8f-a4f14965f103/file///c%3A/Users/muraina/sarif-tutorials/samples/results.sarif'
```

Comment:
Recommend using the minimum inputs for the test. In this case the contents of the `webviewUri` doesn't matter. In particular there's no need to have your username in the string.